### DISCUSSION TITLE
Output comes as mirrored image using SIP Sorcery Image streaming

### DISCUSSION THREAD

**Discussion by jadavparesh06** [sipsorcery-org/sipsorcery#679] [Q&A]
Hello I am using the WebRTC GetStarted example code and instead of the test pattern, I am trying to stream a bitmap image. so before streaming the image shows correctly. but when it gets streamed at client side the result shows mirror effect. means the image is flipped on the client side like how we see it in mirror. whatever is on left shows on right and vice versa.

Any idea what is causing this problem?

**Comment by sipsorcery** [comment]
Mirroring from webcams is common but it's strange for a bitmap. Often with bitmaps you can get a vertical flip due to the way differnt operating systems store them. For example Windows stores bitmaps starting from the bottom right pixel.

I can't help much with what could be causing the problem but perhaps you can use the .NET image [rotateflip function](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.image.rotateflip?view=dotnet-plat-ext-6.0) to deal with it.
